src/subscription.js

```diff
@@ -0,0 +1,278 @@
+    return Q.fcall(function() {
+        var requester = player.getSlackUserFromNameOrID(slack.users, message.user);
+        // TODO: Once we enable WAL - we can remove this lock for this command
+        return db.lock().then(function(unlock) {
+            return db.Subscription.findAll({
+                where: {
+                    requester: requester.name.toLowerCase()
+                },
+                order: [
+                    ['id', 'ASC']
+                ]
+			}).then(function(subscriptions) {
+                var response = "";
+                _.each(subscriptions, function(subscription) {
+                    response += "\nID {id} -> tell {target} when {event} for {source} in {league}".format(subscription.get());
+                });
+
+                unlock.resolve();
+                return response;
+            });
+        });
+    });
+}
+
+//------------------------------------------------------------------------------
+// Processes the remove subscription command
+//
+// subscriptions (by itself) simply lists your subscriptions with ID #s
+//------------------------------------------------------------------------------
+function processRemoveSubscriptionCommand(config, message, id) {
+    return Q.fcall(function() {
+        var requester = player.getSlackUserFromNameOrID(slack.users, message.user);
+        return db.lock().then(function(unlock) {
+            return db.Subscription.findAll({
+                where: {
+                    requester: requester.name.toLowerCase(),
+                    id: id
+                }
+			}).then(function(subscriptions) {
+                if (subscriptions.length === 0) {
+                    unlock.resolve();
+                    return "That is not a valid subscription id";
+                } else if (subscriptions.length === 1) {
+                    return subscriptions[0].destroy().then(function() {
+                        unlock.resolve();
+                        return "Subscription deleted";
+                    });
+                } else {
+                    // This should never happen
+                    unlock.resolve();
+                    throw Error("This should never occur");
+                }
+            });
+        });
+    });
+}
+
+//------------------------------------------------------------------------------
+// Get listeners for a given event and source
+//------------------------------------------------------------------------------
+function getListeners(leagueName, source, event) {
+    // TODO: when we get WAL enabled, we won't need to lock this query any more.
+    return db.lock().then(function(unlock) {
+        return db.Subscription.findAll({
+            where: {
+                league: leagueName.toLowerCase(),
+                source: source.toLowerCase(),
+                event: event.toLowerCase()
+            }
+        }).then(function(subscriptions) {
+            unlock.resolve();
+            var listeners = [];
```

Comment:
one more for the lodash refactor pile:

`return _(subscriptions).map("target").uniq()`